nibabel/openers.py

```diff
@@ -145,3 +145,13 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close_if_mine()
+
+    @classmethod
+    def register_extension(opener_klass, ext, func):
+        """Decorator"""
+        def decorate(klass):
+            assert ext not in opener_klass.compress_ext_map, \
+                "Cannot redefine extension-function mappings."
+            opener_klass.compress_ext_map[ext] = func
```

Comment:
Exactly. The goal was to abstract `Opener` such that new opener methods don't require a new class, and that `Opener` could be used everywhere.

I went this way because `Opener`/`BinOpener` seemed to be used by functions expecting it to know about all file types, so making subclasses of `Opener` when needed, didn't seem like a good option in the end.

But perhaps I'm reading the code poorly... I was looking at `bin/nib-nifti-dx`, `nibabel/arrayproxy.py`, and similar.